recipes/h5pp/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import download, unzip
+import os, re
+
+class h5ppConan(ConanFile):
+    name = "h5pp"
+    version = "1.8.2"
+    description = "A C++17 wrapper for HDF5 with focus on simplicity"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/DavidAce/h5pp"
+    topics = ("conan","h5pp","hdf5", "binary", "storage")
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    requires = "eigen/3.3.7", "spdlog/1.7.0", "hdf5/1.12.0"
+    exports_sources = ["CMakeLists.txt"]
+
+    options = {
+        'tests'     :[True,False],
+        'examples'  :[True,False],
+        'verbose'   :[True,False],
+        }
+
+    default_options = (
+        'tests=False',
+        'examples=False',
+        'verbose=False',
+    )
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        tools.check_min_cppstd(self, "17")
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["H5PP_ENABLE_TESTS"]         = self.options.tests
+            self._cmake.definitions["H5PP_BUILD_EXAMPLES"]       = self.options.examples
+            self._cmake.definitions["H5PP_PRINT_INFO"]           = self.options.verbose
+            self._cmake.definitions["H5PP_DOWNLOAD_METHOD"]      = "conan"
```

Comment:
This will download the dependencies inside cmake.
The cmake script might download different packages as those listed in this recipe's requirements.
Options might differ.
The cmake script should thus use the conan packages of this recipe.

Reply:
If this is just a `if()/endif()` around a `include(conanbuildinfo.cmake)`, then we're fine.
I made this comment because I was recently working on a package that included `conan.cmake`, that called conan itself.
(So its call chain became conan -> cmake -> conan)